include/librpmi.h

```diff
@@ -783,6 +796,230 @@ struct rpmi_service_group {
 	void *priv;
 };
 
+struct rpmi_clock_module;
+
+/** Clock rate match mode */
+enum rpmi_clock_rate_match {
+	RPMI_CLK_RATE_MATCH_INVALID = -1,
+	RPMI_CLK_RATE_MATCH_PLATFORM = 0,
+	RPMI_CLK_RATE_MATCH_ROUND_DOWN = 1,
+	RPMI_CLK_RATE_MATCH_ROUND_UP = 2,
+	RPMI_CLK_RATE_MATCH_MAX_IDX,
+};
+
+/** Supported clock states */
+enum rpmi_clock_state {
+	RPMI_CLK_STATE_INVALID = -1,
+	RPMI_CLK_STATE_DISABLED = 0,
+	RPMI_CLK_STATE_ENABLED = 1,
+	RPMI_CLK_STATE_MAX_IDX,
+};
+
+/** Clock type based on rate format */
+enum rpmi_clock_type {
+	RPMI_CLK_TYPE_INVALID = -1,
+	RPMI_CLK_TYPE_DISCRETE = 0,
+	RPMI_CLK_TYPE_LINEAR = 1,
+	RPMI_CLK_TYPE_MAX_IDX,
+};
+
+/** A clock rate representation in RPMI */
+struct rpmi_clock_rate {
+	rpmi_uint32_t lo;
+	rpmi_uint32_t hi;
+};
```

Comment:
rpmi_clock_rate_match , rpmi_clock_state, rpmi_clock_type , and rpmi_clock_rate should be just after rpmi_clock_service_id  so that it is part of RPMI_MSG_PROTOCOL doxygen group.

Reply:
@pathakraul check this comment.

Reply:
I didnt miss but checked the state of our pdf generated and i think whole librpmi.h will need changes from doxygen perspective so i didnt do anything here

Reply:
[refman.pdf](https://github.com/user-attachments/files/16247604/refman.pdf)

Reply:
I will make the changes for complete librpmi.h and raise a new PR